### PR TITLE
Add geolocate functionality to mapbox-gl-geocoder

### PR DESCRIPTION
**Pull request by krydima**
Add functionality to find user location through browser api.
Screenshot 2021-12-07 at 13 01 29

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] run `npm run docs` and commit changes to API.md
 - [x] update CHANGELOG.md with changes under `master` heading before merging

### CODE REVIEW

**Review by mpothier — APPROVED** [review]
Looks great - a couple minor suggestions for improving code annotations and readability, otherwise LGTM

File: lib/geolocation.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      if (this.supportsGeolocation !== undefined) {
+        res(this.supportsGeolocation);
+    
+      } else if (window.navigator.permissions !== undefined) {
+        // navigator.permissions has incomplete browser support
+        // http://caniuse.com/#feat=permissions-api
+        // Test for the case where a browser disables Geolocation because of an
+        // insecure origin
+        window.navigator.permissions.query({name: 'geolocation'}).then(function (p) {
+          //TODO: PermissionStatus{state: 'denied', onchange: null}
```
Comment (mapbox-danny) [inline lib/geolocation.js:30]: What are your thoughts on messaging the user when navigator permission is denied?

File: lib/index.js [hunk 2]
```diff
@@ -269,6 +291,63 @@ MapboxGeocoder.prototype = {
...
+      this._typeahead.selected = true;
+      this.lastSelected = JSON.stringify(geojson);
+      this._showClearButton();
+      this.fresh = false;
+
+      const config = {
+        limit: 1,
+        language: [this.options.language],
+        query: geojson.geometry.coordinates,
+        types: ["address"]
```
[the remark this answers is not in the document]
Comment (krydima) [reply lib/index.js:314]: User will see "You denied Geolocation".

Screenshot 2021-12-08 at 15 33 09

File: lib/geolocation.js (hunk 1 above)
Comment (krydima) [reply lib/geolocation.js]: User will see "You denied Geolocation".

Screenshot 2021-12-08 at 15 33 09

File: lib/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (krydima) [reply lib/index.js:314]: agreed to leave it as here

File: lib/geolocation.js (hunk 1 above)
Comment (mpothier) [reply lib/geolocation.js]: @krydima @mapbox-danny I'm proposing we change the text to say "Geolocation permission denied"

**Pull request merged**